### DISCUSSION TITLE
How to give different geometrical constraints along the chord of the airfoil?

### DISCUSSION THREAD

**Discussion by usmanshahid90** [mdolab/dafoam#112] [General]
Is it possible to give different geometrical constraints along the chord of the airfoil? 
For example:
1)  for the first 20% of the chord, I give 5% upper and lower bound of thickness.
2) for 20 to 70 % of the chord, i give different thickness and the volume contains and so on?

Or in other words, I am aiming to put weights that which portion of airfoil can be perturbed more and which should be perturbed less. 

Kindly guide me in this regard.
Thanks

**Comment by friedenhe** [comment]
Yes, you can setup two sets of leList and teList and add volume and thickness constraints for the two portions. Something like:

# first portion
DVCon.addVolumeConstraint(leList1, teList1, nSpan=2, nChord=10, lower=1.0, upper=3, scaled=True)
# second portion
DVCon.addVolumeConstraint(leList2, teList2, nSpan=2, nChord=10, lower=1.0, upper=3, scaled=True)

**Reply by usmanshahid90** [reply]
Will first and second portion divides the airfoil into two equal parts? or can divide the airfoil into unequal parts? If in unequal parts then how to define that percentage? Like, i want 30% of my airfoil from the leading edge to have specific constraints. How would I define that 30%?

**Reply by friedenhe** [reply]
It all depends on leList and teList you provide. Read this [page](https://mdolab-mach-aero.readthedocs-hosted.com/en/latest/machAeroTutorials/opt_aero.html#geometric-constraints) regarding how the constraint is done.
